### PR TITLE
S2U-20 Tests & Quizzes: random questions from multiple pools

### PR DESCRIPTION
**Pull request by bgarciaentornos**
Leaving as a draft for the moment. This will be adapted once https://github.com/sakaiproject/sakai/pull/12134 is merged, some of it is based on it but new conflicts will appear on the places where they diverge.

### CODE REVIEW

**Comment by bgarciaentornos** [comment]
I updated the code after rebasing with S2U-19, I will commit a separate commit with some fixes I'm working on atm

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/ItemAuthorBean.java [hunk 1]
```diff
@@ -786,8 +786,8 @@ public List getSectionSelectList() {
       SelectItem selection = new SelectItem();
 
       // need to filter out all the fixed or random draw parts
-      if (part.getSectionAuthorType().equals(SectionDataIfc.RANDOM_DRAW_FROM_QUESTIONPOOL) ||
-          part.getSectionAuthorType().equals(SectionDataIfc.FIXED_AND_RANDOM_DRAW_FROM_QUESTIONPOOL)) {
+      if (SectionDataIfc.RANDOM_DRAW_FROM_QUESTIONPOOL.equals(part.getSectionAuthorType()) || SectionDataIfc.FIXED_AND_RANDOM_DRAW_FROM_QUESTIONPOOL.equals(part.getSectionAuthorType())
+          || SectionDataIfc.RANDOM_DRAW_FROM_QUESTIONPOOLS.equals(part.getSectionAuthorType())) {
```
Comment (stetsche) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/ItemAuthorBean.java:790]:
```suggestion
      if (SectionDataIfc.RANDOM_DRAW_FROM_QUESTIONPOOL.equals(part.getSectionAuthorType()) 
          || SectionDataIfc.FIXED_AND_RANDOM_DRAW_FROM_QUESTIONPOOL.equals(part.getSectionAuthorType())
          || SectionDataIfc.RANDOM_DRAW_FROM_QUESTIONPOOLS.equals(part.getSectionAuthorType())) {
```

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/SectionContentsBean.java [hunk 2]
```diff
@@ -97,7 +97,7 @@ public class SectionContentsBean extends SpringBeanAutowiringSupport implements
   private Integer sectionAuthorType;
   private Integer questionOrdering;
   private Integer numberToBeDrawn;
-  private Long poolIdToBeDrawn;
```
Comment (stetsche) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/SectionContentsBean.java:100]: Why are you changing the access level to protected? The field seems to only be accessed through the public setter.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/SectionBean.java [hunk 3]
```diff
@@ -399,68 +403,37 @@ public List getPoolsAvailable()
...
-            int currItems = delegate.getCountItems(currPool.getQuestionPoolId());
-            if (currItems>0) {
-                resultPoolList.add(new SelectItem((currPool.getQuestionPoolId().toString()), getPoolTitleValueForRandomDrawDropDown(currPool, currItems, allpoollist, delegate)));
-            }
-        } else {
-          // the pool has been deleted
-        }
-    }
 
-    Collections.sort(resultPoolList, new ItemComparator());
```
Comment (jesusmmp) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/SectionBean.java:463]: Don't remove this line

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/SectionContentsBean.java (hunk 2 above)
Comment (bgarciaentornos) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/SectionContentsBean.java:100]: This comes from a previous version, I tried to access it directly but changed my mind.

**Comment by bgarciaentornos** [comment]
Thanks @stetsche and @jesusmmp for your review, I've fixed what you requested

**Review by stetsche — APPROVED** [review]

**Review by jesusmmp — APPROVED** [review]

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages_eu.properties [hunk 4]
```diff
@@ -876,3 +876,6 @@ fixed_and_random_draw_msg_no_date=Parte honetako galderak bi modutan jaso dira:
 edit_published_assessment_warn_edit_pool_fixed_questions=Aldaketak eragina izango dute azterketa honetan soilik, ez erabilitako biltzaileetan. "{0}" izeneko biltzailean dauden galdera finkoak editatzeko sakatu goiko esteka, Galderak, orrialde honetan goian dagoena. Gauza bera egin editatzeko "{1}" izeneko biltzailean dauden galderak.
 notfixedquestionselected_error=Gutxienez galdera finko bat hautatu behar duzu.
 notenough_error=Ez dago galdera nahikorik biltzailean. Galdera finkoen eta ausaz ateratako galderen baturak ez du gainditu behar erabilitako biltzailean daudenak.
+
+# S2U-20
+random_multiple_pools=Random draw from multiple question pools
```
Comment (bgarciaentornos) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages_eu.properties:881]: @pablosanroman could you please translate this?

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages_eu.properties [hunk 5]
```diff
@@ -514,9 +514,9 @@ options=Aukerak
 type=Mota
 type_onebyone=Galderak banan-banan eginak
 random_draw_type=Ausazko bista hemendik\:
-random_draw_msg=Parte honetako galderak ausaz sortzen dira galdera-biltzaile batetik. Egin klik Ebaluazioaren aurrebistan adibide bat ikusteko. 
-random_draw_msg_published=Parte honetako galderak galderen gordailuak sortu ditu, <b>{0}</b>, on <b>{1}</b> at <b>{2}</b>. Parte honetako galderak editatzeko, egin klik hemen\: <b>Editatu galderak</b>.
-random_draw_msg_no_date=Parte honetako galderak ausaz sortzen dira galdera gordailu batetik. Egin klik \u2018Aurreko azterketa\u2019 dioen tokian lagin bat ausaz ateratzeko.
+random_draw_msg=Parte honetako galderak ausaz sortzen dira galdera-biltzaile batetik . Egin klik Ebaluazioaren aurrebistan adibide bat ikusteko. 
+random_draw_msg_published=Parte honetako galderak galderen gordailuak sortu ditu , <b>{0}</b>, on <b>{1}</b> at <b>{2}</b>. Parte honetako galderak editatzeko, egin klik hemen\: <b>Editatu galderak</b>.
+random_draw_msg_no_date=Parte honetako galderak ausaz sortzen dira galdera gordailu batetik . Egin klik \u2018Aurreko azterketa\u2019 dioen tokian lagin bat ausaz ateratzeko.
```
Comment (bgarciaentornos) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages_eu.properties:519]: @pablosanroman Could you please add the changes to these lines?

I copy here the modified Spanish ones:
random_draw_msg=Las preguntas de esta parte son generadas aleatoriamente a partir de una(s) bater\u00eda(s) de preguntas, <b>{0}</b>, en <b>{1}</b> a <b>{2}</b>. Para obtener los \u00faltimos cambios pulsa en <b>Actualizar preguntas</b> o pulsa en <b>Editar</b> de la parte y finalmente en <b>Guardar</b>. Pulsando en <b>Vista previa</b> o en <b>Publicar</b> se actualizar\u00e1 autom\u00e1ticamente el grupo de preguntas al igual que al exportar un examen.
random_draw_msg_published=Las preguntas de esta parte se generaron desde la(s) bater\u00eda(s) de preguntas, <b>{0}</b>, en <b>{1}</b> en <b>{2}</b>. Para editar las preguntas en esta parte, haga clic en <b>Editar preguntas</b>.
random_draw_msg_no_date=Las preguntas de esta parte son generadas aleatoriamente a partir de una(s) bater\u00eda(s) de preguntas. Haga clic en Vista previa para ver un ejemplo.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages_eu.properties (hunk 4 above)
Comment (PabloSanRoman) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages_eu.properties:881]: Hi @bgarciaentornos !!
random_multiple_pools=Ausazko bista biltzaileetatik

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages_eu.properties (hunk 5 above)
Comment (PabloSanRoman) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages_eu.properties:519]: Hi @bgarciaentornos , here they are: 

random_draw_msg=Parte honetako galderak ausaz jasota daude biltzaile batetik edo hainbatetik,  <b>{0}</b>, on <b>{1}</b> at <b>{2}</b>. Sakatu "Aurrebista" adibide bat ikusteko. Azken aldaketak lortzeko sakatu "Eguneratu galderak" edo sakatu partearen "Editatu" eta gero "Gorde". "Aurrebista" edo "Argitaratu" sakatuz gero, automatikoki eguneratuko da galdera-multzoa, azterketa esportatzean bezalaxe.

random_draw_msg_published=Parte honetako galderak {0} biltzaileetatik jasota daude: <b>{0}</b>, on <b>{1}</b> at <b>{2}</b>. Parte honetako galderak editatzeko sakatu "Editatu galderak" botoia.

random_draw_msg_no_date=Parte honetako galderak ausaz jasota daude biltzaile batetik edo hainbatetik. Sakatu "Aurrebista" adibide bat ikusteko.

**Comment by mpellicer** [comment]
Thank you very much guys @bgarciaentornos @stetsche @jesusmmp , collaboration was key!

**Pull request merged**